src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -927,7 +931,9 @@ bool AsyncRPCOperation_sendmany::load_inputs(TxValues& txValues) {
 bool AsyncRPCOperation_sendmany::find_unspent_notes() {
     std::vector<SproutNoteEntry> sproutEntries;
     std::vector<SaplingNoteEntry> saplingEntries;
-    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, fromaddress_, mindepth_);
+    // TODO: move this to the caller
+    auto zaddr = KeyIO(Params()).DecodePaymentAddress(fromaddress_);
+    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, zaddr, mindepth_);
```

Comment:
This is conflating usages of `std::optional`:
- The return value of `DecodePaymentAddress` uses `std::nullopt` to mean "invalid address".
- The parameter of `GetFilteredNotes` uses `std::nullopt` to mean "don't filter notes".

I don't think this matters, because `fromaddress_` has already been validated in the `z_sendmany` RPC logic by now IIRC (and also I'm told this will change in a later commit).